### PR TITLE
JSON RPC Client cleanup

### PR DESCRIPTION
**Pull request by hackaugusto**
removed unused methods / properties:
- new_abi_contract
- coinbase
- find_block

removed patches:
- patch_send_transaction
- patch_send_message

### CODE REVIEW

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -508,9 +428,15 @@ def __init__(self, sender, abi, address, call_func, transact_func, estimate_func
 
 class JSONRPCClient(object):
 
-    def __init__(self, host, port, privkey):
+    def __init__(self, host, port, privkey, nonce_update_interval=5.0, nonce_offset=0):
+        endpoint = 'http://{}:{}'.format(host, port)
+        session = requests.Session()
+        adapter = requests.adapters.HTTPAdapter(pool_maxsize=50)
```
Comment (palango) [inline raiden/network/rpc/client.py]: Maybe make pool_maxsize an optional parameter as well and add a docstring.

File: raiden/network/rpc/client.py [hunk 2]
```diff
@@ -519,14 +445,15 @@ def __init__(self, host, port, privkey):
...
+        self.nonce_current_value = None
+        self.nonce_lock = Semaphore()
+        self.nonce_update_interval = nonce_update_interval
+        self.nonce_offset = nonce_offset
+
     def __repr__(self):
         return '<JSONRPCClient @%d>' % self.port
 
-    @property
-    def coinbase(self):
```
Comment (palango) [inline raiden/network/rpc/client.py]: Should this and `find_block` be removed or should they stay in the client. Not sure if the will be used/are useful though.

Comment (hackaugusto) [reply raiden/network/rpc/client.py]: Raiden signs all transactions locally, we don't need coinbase ATM.

Also, it's not hard to add these back if the need arises, right now it's just code bloat.

Comment (palango) [reply raiden/network/rpc/client.py]: Fine for me, can you add some docs to the `__init__` method, then it's ready for merging.

**Pull request merged**